compiler/rustc_trait_selection/src/traits/query/normalize.rs

```diff
@@ -207,13 +206,10 @@ impl<'cx, 'tcx> FallibleTypeFolder<'tcx> for QueryNormalizer<'cx, 'tcx> {
                         let substs = substs.try_fold_with(self)?;
                         let recursion_limit = self.tcx().recursion_limit();
                         if !recursion_limit.value_within_limit(self.anon_depth) {
-                            let obligation = Obligation::with_depth(
-                                self.cause.clone(),
-                                recursion_limit.0,
-                                self.param_env,
-                                ty,
-                            );
-                            self.infcx.err_ctxt().report_overflow_error(&obligation, true);
+                            // We have been recursive without limits.  Since we want to make
+                            // progress without halting compilation, fall back to the worse path of
+                            // not revealing the opaques.
+                            return ty.try_super_fold_with(self);
```

Comment:
This makes me nervous. Why is necessary to avoid reporting an overflow here?

Reply:
Changed to a comment + delayed bug.